api/v3/utils.php

```diff
@@ -2472,12 +2472,42 @@ function _civicrm_api3_field_value_check(&$params, $fieldName, $type = NULL) {
  */
 function _civicrm_api3_basic_array_get($entity, $params, $records, $idCol, $filterableFields) {
   $options = _civicrm_api3_get_options_from_params($params, TRUE, $entity, 'get');
-  // TODO // $sort = CRM_Utils_Array::value('sort', $options, NULL);
   $offset = $options['offset'] ?? NULL;
   $limit = $options['limit'] ?? NULL;
 
+  $sort = !empty($options['sort']) ? explode(', ', $options['sort']) : NULL;
+  if ($sort) {
+    usort($records, function($a, $b) use ($sort) {
+      foreach ($sort as $field) {
+        [$field, $dir] = array_pad(explode(' ', $field), 2, 'asc');
+        $modifier = strtolower($dir) == 'asc' ? 1 : -1;
+        if (isset($a[$field]) && isset($b[$field])) {
+          if ($a[$field] == $b[$field]) {
+            continue;
+          }
+          return (strnatcasecmp($a[$field], $b[$field]) * $modifier);
+        }
+        elseif (isset($a[$field]) || isset($b[$field])) {
+          return ((isset($a[$field]) ? 1 : -1) * $modifier);
+        }
+      }
+      return 0;
+    });
+  }
+
   $matches = [];
 
+  $isMatch = function($record, $key, $searchValue) {
+    if (is_array($searchValue) && count($searchValue) === 1 && in_array(array_keys($searchValue)[0], CRM_Core_DAO::acceptedSQLOperators())) {
+      $operator = array_keys($searchValue)[0];
+      $searchValue = array_values($searchValue)[0];
+      return \Civi\Api4\Generic\Traits\ArrayQueryActionTrait::filterCompare($record, [$key, $operator, $searchValue]);
```

Comment:
I don't love converting this to static - could we instantiate an instance & call it with $this-> ?

Reply:
Not really, no. It's a trait. And traits can't be instantiated, nor can they be used by classless APIv3 files.

Reply:
One alternative would be copy & paste, if you're worried about future APIv4 refactoring affecting v3. But IMO we can cross that bridge later & this is ok for now.

Reply:
@colemanw honestly I'd prefer copy & paste - it's kinda contaminating the structure of what we are building for apiv4 with the needs of our legacy api